platform/android/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/offline/ChangeResourcesCachePathActivity.kt

```diff
@@ -0,0 +1,136 @@
+package com.mapbox.mapboxsdk.testapp.activity.offline
+
+import android.annotation.TargetApi
+import android.content.Context
+import android.os.Build
+import android.os.Bundle
+import android.support.v7.app.AppCompatActivity
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.AdapterView
+import android.widget.BaseAdapter
+import android.widget.TextView
+import android.widget.Toast
+import com.mapbox.mapboxsdk.storage.FileSource
+import com.mapbox.mapboxsdk.testapp.R
+import kotlinx.android.synthetic.main.activity_change_resources_cache_path.*
+import java.io.File
+import java.util.ArrayList
+
+class ChangeResourcesCachePathActivity : AppCompatActivity(),
+    AdapterView.OnItemClickListener,
+    FileSource.SetResourcesCachePathCallback {
+
+  lateinit var adapter: PathAdapter
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_change_resources_cache_path)
+
+    Thread(Runnable {
+      adapter = PathAdapter(this, obtainExternalFilesPaths(this))
+      listView.adapter = adapter
+      listView.emptyView = empty
+      listView.onItemClickListener = this
+    }).start()
+  }
+
+  override fun onStart() {
+    super.onStart()
+    val path: String? = FileSource.getResourcesCachePath(this)
+    Toast.makeText(this, "Current path: $path", Toast.LENGTH_LONG).show()
+  }
+
+  override fun onItemClick(parent: AdapterView<*>?, view: View?, position: Int, id: Long) {
+    listView.onItemClickListener = null
+    val path: String = adapter.getItem(position) as String
+    FileSource.setResourcesCachePath(this, path, this)
+  }
+
+  override fun onError(message: String?) {
+    listView.onItemClickListener = this
+    Toast.makeText(this, "Error: $message", Toast.LENGTH_LONG).show()
+  }
+
+  override fun onSuccess(path: String?) {
+    listView.onItemClickListener = this
+    Toast.makeText(this, "New path: $path", Toast.LENGTH_LONG).show()
+  }
+
+  private fun obtainExternalFilesPaths(context: Context): List<String> {
```

Comment:
It'd be great if this method returned one additional hardcoded path to internal storage for test purposes when the device doesn't have external storage capabilities.

Reply:
Added the internal path and an invalid path.